aii-pxelinux/src/main/perl/pxelinux.pm

```diff
@@ -374,19 +402,47 @@ EOF
     $fh->close();
 }
 
-# Prints an IP address in hexadecimal.
-sub hexip
+
+# Write the Grub2 configuration file.
+# TODO: handle append options?
+sub write_grub2_config
 {
-    my $ip = shift || "";
+    my $cfg = shift;
+    my $t = $cfg->getElement (PXEROOT)->getTree;
+    my $fh = CAF::FileWriter->open (filepath ($cfg, PXE_VARIANT_GRUB2),
+                                    log => $this_app, mode => 0644);
 
-    return sprintf ("%02X%02X%02X%02X", $1, $2, $3, $4) if ($ip =~ m/^(\d+)\.(\d+)\.(\d+)\.(\d+)$/);
+    my $linux_cmd = $this_app->option(GRUB2_EFI_LINUX_CMD);
+    unless ( $linux_cmd ) {
+        $this_app->error("AII option ".GRUB2_EFI_LINUX_CMD." undefined");
```

Comment:
I am not sure to understand what is the problem? As far as I know, `pxelinux` is called once for every host (and not once with a list of host. That means that you'll get it once for every host passed to `aii-shellfe` command.

I moved the error before opening the file and the function now returns a different status whether the file was updated or not.